db/commitlog/commitlog.cc

```diff
@@ -1432,41 +1497,360 @@ future<R> db::commitlog::segment_manager::allocate_when_possible(T writer, db::t
     // If this is already too big now, we should throw early. It's also a correctness issue, since
     // if we are too big at this moment we'll never reach allocate() to actually throw at that
     // point.
-    sanity_check_size(size);
+    if (size < max_mutation_size) {
```

Comment:
Let's drop max_mutation_size, and instead fragment whenever the mutation is >= 1k and doesn't fit in the segment tail. Just to get this code exercised more often (also waste less commitlog space).

Reply:
Effectively it renames max_mutation_size to max_unfragmented_mutation_size and makes it smaller.

Reply:
That is not a good idea, because the fragmented path is _significantly_ slower and also requires blocking out any other "parallel" write to the commit log.

Reply:
Ok.